### PR TITLE
Add Navigation drawer with tags list

### PR DESCRIPTION
**Pull request by kvj**
As per discussion, I extracted navigation drawer with tags list into separate pull request. At present moment click on tag does nothing

Screenshot
screenshot_20170211-085848

### CODE REVIEW

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -18,4 +18,17 @@
 
     <style name="AppTheme.PopupOverlay" parent="ThemeOverlay.AppCompat.Light" />
 
+    <style name="AppTheme.DrawerEmail">
+        <item name="android:fontFamily">Roboto Medium</item>
+        <item name="android:textSize">14sp</item>
+        <item name="android:singleLine">true</item>
```
Comment (nbau21) [inline app/src/main/res/values/styles.xml]: use `<item name="android:maxLines">1</item>`.

`singleLine` is deprecated

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -18,4 +18,17 @@
...
+        <item name="android:ellipsize">middle</item>
+    </style>
+    <style name="AppTheme.DrawerServer">
+        <item name="android:fontFamily">Roboto Normal</item>
+        <item name="android:textSize">12sp</item>
+        <item name="android:singleLine">true</item>
```
Comment (nbau21) [inline app/src/main/res/values/styles.xml]: use `<item name="android:maxLines">1</item>`.

singleLine is deprecated

File: app/src/main/res/menu/drawer_tags.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<menu xmlns:android="http://schemas.android.com/apk/res/android">
+    <item android:title="Tags" android:id="@+id/menu_account_tags">
+        <menu/>
```
Comment (nbau21) [inline app/src/main/res/menu/drawer_tags.xml]: unused  menu tag?

File: app/src/main/res/layout/view_navigation_header.xml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+                android:id="@+id/main_account_email"
+                style="@style/AppTheme.DrawerEmail"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:text=""/>
+            <TextView
+                android:id="@+id/main_account_server"
+                style="@style/AppTheme.DrawerServer"
+                android:layout_width="match_parent"
+                android:text=""
```
Comment (nbau21) [inline app/src/main/res/layout/view_navigation_header.xml]: if empty, don't set text to "". just delete this line

File: app/src/main/res/layout/view_navigation_header.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+            android:layout_width="0dp"
+            android:layout_gravity="center_vertical"
+            android:layout_weight="1"
+            android:layout_height="wrap_content">
+            <TextView
+                android:id="@+id/main_account_email"
+                style="@style/AppTheme.DrawerEmail"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:text=""/>
```
Comment (nbau21) [inline app/src/main/res/layout/view_navigation_header.xml]: if empty, don't set text to "". just delete this line

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -1,40 +1,51 @@
...
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:layout_gravity="bottom|end"
-        android:layout_margin="@dimen/fab_margin"
-        app:srcCompat="@drawable/fab_add" />
+    </android.support.design.widget.CoordinatorLayout>
+    <android.support.design.widget.NavigationView
+        android:id="@+id/drawer"
+        android:layout_gravity="start"
+        android:layout_width="260dp"
```
Comment (nbau21) [inline app/src/main/res/layout/activity_main.xml]: don't hardcode width, use `android:layout_width="wrap_content"`

File: app/src/main/java/org/standardnotes/notes/store/ValueStore.kt [hunk 7]
```diff
@@ -22,6 +22,10 @@ class ValueStore(context: Context) {
         get() = prefs.getString("server", BuildConfig.SERVER_DEFAULT)
         set(value) { prefs.edit().putString("server", value).apply() }
 
+    var email: String?
+        get() = prefs.getString("email", "<Not set>") // Will never happen on production
```
Comment (nbau21) [inline app/src/main/java/org/standardnotes/notes/store/ValueStore.kt]: just default as `""`.

File: app/src/main/java/org/standardnotes/notes/MainActivity.kt [hunk 8]
```diff
@@ -47,11 +85,22 @@ class MainActivity : BaseActivity() {
     }
 
     override fun onOptionsItemSelected(item: MenuItem?): Boolean {
+        if (drawerToggle!!.onOptionsItemSelected(item))
+            return true
```
Comment (nbau21) [inline app/src/main/java/org/standardnotes/notes/MainActivity.kt]: add else after the `if (drawerToggle...`.

File: app/src/main/res/menu/drawer_tags.xml (hunk 3 above)
Comment (kvj) [inline app/src/main/res/menu/drawer_tags.xml]: No, tags list is submenu of menu_account_tags menu item

File: app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (kvj) [inline app/src/main/res/layout/activity_main.xml]: No, wrap_content will not work here. Width of popup drawer should be fixed. I'll move it's value to dimen.xml

File: app/src/main/java/org/standardnotes/notes/MainActivity.kt (hunk 8 above)
Comment (kvj) [inline app/src/main/java/org/standardnotes/notes/MainActivity.kt]: Added

File: app/src/main/java/org/standardnotes/notes/store/ValueStore.kt (hunk 7 above)
Comment (kvj) [inline app/src/main/java/org/standardnotes/notes/store/ValueStore.kt]: Fixed

File: app/src/main/res/layout/view_navigation_header.xml (hunk 4 above)
Comment (kvj) [inline app/src/main/res/layout/view_navigation_header.xml]: Deleted

File: app/src/main/res/layout/view_navigation_header.xml (hunk 5 above)
Comment (kvj) [inline app/src/main/res/layout/view_navigation_header.xml]: Deleted

File: app/src/main/res/values/styles.xml (hunk 1 above)
Comment (kvj) [inline app/src/main/res/values/styles.xml]: Fixed

File: app/src/main/res/values/styles.xml (hunk 2 above)
Comment (kvj) [inline app/src/main/res/values/styles.xml]: Fixed

**Comment by deftelf** [comment]
Lovely, ta.

**Pull request merged**